tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/reindexing/ElasticsearchReindexingTests.java

```diff
@@ -360,10 +366,16 @@ private void cleanUpIndex()
     {
         try
         {
-            DeleteByQueryRequest request = new DeleteByQueryRequest(CUSTOM_ALFRESCO_INDEX);
-            request.setQuery(QueryBuilders.matchAllQuery());
-            BulkByScrollResponse response = elasticClient.deleteByQuery(request, RequestOptions.DEFAULT);
-            STEP("Deleted " + response.getDeleted() + " documents from index");
+            DeleteByQueryRequest request = new DeleteByQueryRequest.Builder().index(CUSTOM_ALFRESCO_INDEX)
+                        .refresh(true)
+                        .conflicts(Conflicts.Proceed)
```

Comment:
Instead of setting an option to `proceed` in a case when a query operation runs into a version conflict, I would recommend implementing Retry mechanism to avoid conflicts.
The Rest High client has the retry mechanism implemented, see: https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/index/reindex/AbstractBulkByScrollRequest.java#L83
Sadly, `opensearch-java` doesn't have such a mechanism.

Reply:
Added the change to Refresh the index before DeleteByQuery